### ISSUE TITLE
msgpack reads python 2 string as bytes

### ISSUE THREAD

**Issue by chris-b1** [pandas-dev/pandas#12225]
This maybe is just a doc warning - but caught me by surprise.  I would prefer the `read_csv` behavior, but it would be backwards incompatible.

``` python
In [57]: pd.read_csv('path/doesn_exist')
---------------------------------------------------------------------------

IOError: File path/doesn_exist does not exist

In [56]: pd.read_msgpack('path/doesn_exist')
Out[56]: [112, 97, 116, 104, 47, 100, 111, 101, 115, 110, 95, 101, 120, 105, 115, 116]

In [59]: pd.read_msgpack(u'path/doesn_exist')
---------------------------------------------------------------------------
ValueError: path_or_buf needs to be a string file path or file-like
```

**Comment by jreback** [comment]
see #12142

**Comment by jreback** [comment]
xref https://github.com/pydata/pandas/issues/5924

**Comment by chris-b1** [comment]
Closed in favor of #15296
